google-cloud-tools-plugin/jps-plugin/src/com/google/cloud/tools/intellij/jps/GenRepoInfoFileModuleBuilder.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.jps;
+
+import com.google.cloud.tools.appengine.api.debug.DefaultGenRepoInfoFileConfiguration;
+import com.google.cloud.tools.appengine.api.debug.GenRepoInfoFile;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkGenRepoInfoFile;
+import com.google.cloud.tools.appengine.cloudsdk.internal.process.ExitCodeRecorderProcessExitListener;
+import com.google.cloud.tools.intellij.jps.model.JpsStackdriverModuleExtension;
+import com.google.cloud.tools.intellij.jps.model.impl.JpsStackdriverModuleExtensionImpl;
+import com.google.common.annotations.VisibleForTesting;
+
+import com.intellij.openapi.diagnostic.Logger;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.jps.ModuleChunk;
+import org.jetbrains.jps.builders.DirtyFilesHolder;
+import org.jetbrains.jps.builders.java.JavaModuleBuildTargetType;
+import org.jetbrains.jps.builders.java.JavaSourceRootDescriptor;
+import org.jetbrains.jps.incremental.BuilderCategory;
+import org.jetbrains.jps.incremental.CompileContext;
+import org.jetbrains.jps.incremental.ModuleBuildTarget;
+import org.jetbrains.jps.incremental.ModuleLevelBuilder;
+import org.jetbrains.jps.incremental.ProjectBuildException;
+import org.jetbrains.jps.model.module.JpsModule;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+
+/**
+ * Generates source context files for a module using the Cloud SDK.
+ */
+public class GenRepoInfoFileModuleBuilder extends ModuleLevelBuilder {
+
+  private static final Logger LOG =
+      Logger.getInstance("#com.google.cloud.tools.intellij.jps.GenRepoInfoFileModuleBuilder");
+
+  public static final String NAME = "Stackdriver source context generator";
+  private GenRepoInfoFileActionFactory actionFactory = new GenRepoInfoFileActionFactory();
+
+  public GenRepoInfoFileModuleBuilder() {
+    super(BuilderCategory.CLASS_POST_PROCESSOR);
+  }
+
+  @VisibleForTesting
+  GenRepoInfoFileModuleBuilder(GenRepoInfoFileActionFactory actionFactory) {
```

Comment:
GenRepoInfoFileModuleBuilder() can't call `this(new GenRepoInfoFileActionFactory());` because `GenRepoInfoFileActionFactory` can't be referenced until the supertype constructor is called.

I went ahead and duplicated the `super(BuilderCategory.CLASS_POST_PROCESSOR);` call in both constructors. LMKWYT